tests/Core/Ruleset/ShowSniffDeprecationsTest.php

```diff
@@ -153,7 +153,7 @@ public static function dataDeprecatedSniffsListDoesNotShowNeedsCsMode()
         return [
             'Standard using deprecated sniffs; documentation is requested' => [
                 'standard'       => __DIR__.'/ShowSniffDeprecationsTest.xml',
-                'additionalArgs' => ['--generator=text'],
+                'additionalArgs' => ['--generator=Text'],
```

Comment:
Please undo this change. This is perfectly valid. Your PR needs a fix, not this unrelated test.

Reply:
> Eh.. the test failure is not unrelated...

Indeed, I chose a poor word to describe what I meant.

> Please undo this change. This is perfectly valid. Your PR needs a fix, not this unrelated test.

Just checking, did you check the commit message? Do you disagree with my assessment describing why I updated the test?

Reply:
I did not, but have now and I don't agree with your assessment.

The basic premise for the CLI arguments is that they should be handled case-insensitively, so `text` is perfectly valid and should be accepted.

Reply:
But on `master`, `phpcs --standard=Generic --generator=text` does not work and throws a fatal error. Or maybe it throws a fatal error on Linux (case-sensitive OS) but not on Windows (case-insensitive OS)?

If that is the case, I failed to consider case-insensitive operating systems in this PR. I'm not sure what would be the best approach for this PR to accommodate this.

Maybe perform a case-insensitive check for the generator name when running on Windows and a case-sensitive check when running on other systems? What I don't like about this approach is that PHPCS would continue to behave differently depending on the OS.

Another option is to change PHPCS to ensure that it behaves consistently across different operating systems and handles the `--generator` CLI argument case-insensitively. 

> The basic premise for the CLI arguments is that they should be handled case-insensitively

I was not aware of that and I can think of one other example of a CLI argument that is not case-insensitive when PHPCS is running on Linux: `--standard=generic`. I'm not sure if there are other cases.

```
phpcs --generator=Text --standard=generic                                              
ERROR: the "generic" coding standard is not installed. The installed coding standards are MySource, PEAR, PSR1, PSR2, PSR12, Squiz, Zend and MyStandard
```

Reply:
> But on `master`, `phpcs --standard=Generic --generator=text` does not work and throws a fatal error. Or maybe it throws a fatal error on Linux (case-sensitive OS) but not on Windows (case-insensitive OS)?

Exactly. It is currently (unintentionally) OS-dependent, but shouldn't be.

> Another option is to change PHPCS to ensure that it behaves consistently across different operating systems and handles the `--generator` CLI argument case-insensitively.

IMO, that would be the correct thing to do.

> > The basic premise for the CLI arguments is that they should be handled case-insensitively
> 
> I was not aware of that and I can think of one other example of a CLI argument that is not case-insensitive when PHPCS is running on Linux: `--standard=generic`. I'm not sure if there are other cases.

That's outside the scope of this PR. Might be worth opening an issue to investigate.

Reply:
Re: CLI and case-insensitive: https://github.com/squizlabs/PHP_CodeSniffer/pull/1656#issuecomment-331772466

Reply:
Thanks for the additional information. I will drop the commit that changed `testDeprecatedSniffsListDoesNotShowNeedsCsMode()` and update this PR to ensure that the error displayed in `Config::processLongArgument()` for invalid generator names treats the value of the `--generator` argument case-insensitively.

> > Another option is to change PHPCS to ensure that it behaves consistently across different operating systems and handles the --generator CLI argument case-insensitively.
>
> IMO, that would be the correct thing to do.

To implement this, I believe it will be necessary to change `Runner::runPHPCS()` to load the correct generator class no matter what is the case of the passed generator argument. This is a separate change not directly related to #709. Do you prefer to review this change in this PR or a separate PR?

Reply:
Why do you think the case-(in)sensitively should be handled in the `Runner` class instead of in the `Config` class ?

Reply:
Because I did not consider normalizing the input in the `Config` class when setting `Config::generator`, and was thinking about doing that in the `Runner` class.

To illustrate what I had in mind, assuming `--generator=text` is passed, my plan was to do a case-insensitive check in the `Config` class to see if the generator name passed is valid. Then, on the `Runner` class the code normalizes the generator name (i.e, modifies the generator name to match the case of the desired file/class name).

After seeing your question, I realized that it makes sense to normalize the generator name in the `Config` class, and that is probably what you had in mind. Is this correct?

I was also missing that `Runner::runPHPCS()` creates an instance of `Config`. Since I was missing that, I was worried about `runPHPCS()` being called directly and attempting to instantiate a generator without the validation and normalization performed in the `Config` class.